graylog2-web-interface/src/components/configurations/UrlWhiteListForm.jsx

```diff
@@ -0,0 +1,219 @@
+  const _onAdd = (event: Event) => {
+    event.preventDefault();
+    setConfig({ ...config, entries: [...config.entries, { id: uuid(), title: '', value: '', type: literal }] });
+  };
+
+  const _onRemove = (event: MouseEvent, idx: number) => {
+    event.preventDefault();
+    // eslint-disable-next-line prefer-const
+    let stateUpdate = cloneDeep(config);
+    const validationUpdate = cloneDeep(validationState);
+    validationUpdate.errors[idx] = null;
+    setValidationState(validationUpdate);
+    stateUpdate.entries.splice(idx, 1);
+    setConfig(stateUpdate);
+  };
+
+  const validURL = (str: string) => {
+    let isValid = true;
+    try {
+      // eslint-disable-next-line no-unused-vars
+      const test = new URL(str);
+    } catch (e) {
+      isValid = false;
+    }
+    return isValid;
+  };
+
+  const _isFormValid = (): boolean => {
+    let isValid = true;
+    if (validationState.errors.length > 0
+      && validationState.errors.find((el => (el && el.title && el.title.valid) === false
+      || (el && el.value && el.value.valid === false)))) {
+      isValid = false;
+    }
+    return isValid;
+  };
+
+  const _updateState = (idx: number, type: string, name: string, value: string) => {
+    const stateUpdate = cloneDeep(config);
+    stateUpdate.entries[idx][name] = value;
+    stateUpdate.entries[idx] = { ...stateUpdate.entries[idx], type };
+    setConfig(stateUpdate);
+  };
+
+  const _updateValidationError = (idx: number, type: string, name: string, result: Object, value: string) => {
+    const validationUpdate = cloneDeep(validationState);
+    validationUpdate.errors[idx] = { ...validationUpdate.errors[idx], [name]: result };
+    setValidationState(validationUpdate);
+    _updateState(idx, type, name, value);
+  };
+
+  const _validate = (name: string, idx: number, type: string, value: string): void => {
+    switch (name) {
+      case 'title': {
+        const result = value.trim().length <= 0 ? { valid: false } : { valid: true };
+        _updateValidationError(idx, type, name, result, value);
+      }
+        break;
+      case 'value':
+        if (type === literal) {
+          const result = validURL(value) ? { valid: true } : { valid: false };
+          _updateValidationError(idx, type, name, result, value);
+        } else if (type === regex && value.trim().length > 0) {
+          const promise = ToolsStore.testRegexValidity(value);
+          promise.then((result) => {
+            const res = result.is_valid ? { valid: true } : { valid: false };
+            _updateValidationError(idx, type, name, res, value);
+          });
+        } else {
+          const res = { valid: false };
+          _updateValidationError(idx, type, name, res, value);
+        }
+        break;
+      default:
+        break;
+    }
+  };
+
+  const _onInputChange = (event: SyntheticInputEvent<EventTarget>, idx: number, type: string) => {
+    if (type === regex) {
+      const debouncedValidate = debounce(_validate, 500);
```

Comment:
This unfortunately doesn't work, since the function is created all the time. This is what happens when I look at the network output in the browser console:
![Kapture 2019-12-20 at 13 06 12](https://user-images.githubusercontent.com/716185/71253922-8f1e9980-2329-11ea-8a3d-a9b680c66976.gif)

Reply:
Should be fixed now !